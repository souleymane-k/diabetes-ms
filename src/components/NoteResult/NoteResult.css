.Result {
    background-color: white;
    color: black;
    margin: 15px;
    max-width: 280px;
    padding: 8px 0 8px 8px;
    border: solid;
    border-radius: 10px;
    /* position: relative; */
  }
  h2{
      text-align: center;
  }
  
  .Result_delete{
    background-color:green;
    border: 0;
    color:white;
    font-size: 12px;
    width:60px;
    margin: 10px;
  }
  .Result_delete:hover {
    background-color: #bd0909;
    color: white;
    cursor: pointer;
    font-weight: bold;
  }

  /* span{
      border:solid;
  } */

  
  /* .Result__result_read{
    font-size: 20px;
    margin: 0;
    max-width: 80%;
    min-height: 24px;
  } */
  /* .Result__result_read a {
    color: #cacaca;
    display: block;
    text-decoration: none;
  } */
  /* .Result__result_read a:hover {
    color: #FFB72F;
  } */
  
  /* .Result__date_tested {
    bottom: 8px;
    color: #a7a5a5;
    font-size: 14px;
    position: absolute;
  } */