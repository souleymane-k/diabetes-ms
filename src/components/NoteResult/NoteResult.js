import React from 'react'
 import { Link } from 'react-router-dom'
import ApiContext from '../../contexts/ApiContext';
import TokenService from '../../services/TokenService'
import config  from '../../config.js'
import './NoteResult.css'
import moment from 'moment'

export default class NoteResult extends React.Component {
  static defaultProps ={
    month_taken: new Date (),
    date_tested: new Date(),
    onDeleteResult: () => {},
  }
  static contextType = ApiContext;

  handleClickDelete = e => {
    e.preventDefault()
    const id = this.props.id

    fetch(`${config.API_ENDPOINT}/results/${id}` , {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json',
        'authorization':`bearer ${TokenService.getAuthToken()}`
      },
    })
      .then(() => {
        this.props.onDeleteResult(id)
      })
      .catch(error => {
        console.log(e)
        console.error({ error })
      })
  }


  render() {
    
    const {id,
      month_taken,
      meal_taken,result_read,date_tested,description,diabetestype} = this.props
    
    return (
      <div className="result-form">
      <div className='Result'>
        <h2 className='Result_link'>
        <Link to={`/AddResult/${id}`}>
             ADD RESULT
          </Link> 
        </h2>
        <div className="resultElements">
        <div className='Result_month_taken'>
          <div className='Result_results_description'>
          month of

          {'  '}

            <span className='month_taken'>
              {moment.utc(month_taken).format("MMMM")}
            </span> 
          </div> 
        
         </div>
         <div className='Result__meal_taken'>
          <div className='Resultresults_meal_taken'>
          meal
          {'   '}
            <span className='meal_taken'>
              {meal_taken}
            </span> 
          </div> 
         </div>
         <div className='Result_result_read'>
          <div className='Result_result_read'>
          result
          {'   '}
            <span className='Result_read'>
              {result_read} {'   '} Mg/Dl
            </span> 
          </div> 
         </div>
         <div className='Result_result_dates'>
          <div className='Result_result_date__tested'>
          date
          {'   '}
            <span className='date_tested'>
              {moment.utc(date_tested).format("MM/DD/YYYY")}
            </span> 
          </div> 
         </div>
         <div className='Results_description'>
          <div className='Result_results_description'>
          Description
            {'   '}
            <span className='Description'>
              {description}
            </span> 
          </div> 
         </div>
         <div className='Result_diabetestype'>
          <div className='Result_results_diabetestype'>
          diabetestype
          {'   '}
            <span className='Result_diabetestype'>
              {diabetestype}
            </span> 
          </div> 
         </div>
         <button
          className='Result_delete'
          type='button'
          onClick={this.handleClickDelete}
        >
          DELETE
        </button>
        <Link  to={`/editresult/${id}`} className="edit_result">Edit Result</Link>
      </div>
      </div>
      </div>

    )
  }
}