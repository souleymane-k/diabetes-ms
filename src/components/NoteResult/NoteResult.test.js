import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NoteResult from './NoteResult';

describe(`NoteResult component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<NoteResult />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NoteResult />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})