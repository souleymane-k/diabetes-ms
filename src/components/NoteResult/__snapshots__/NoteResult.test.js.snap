// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteResult component renders the complete form 1`] = `
<div
  className="result-form"
>
  <div
    className="Result"
  >
    <h2
      className="Result_link"
    >
      <Link
        to="/AddResult/undefined"
      >
        ADD RESULT
      </Link>
    </h2>
    <div
      className="resultElements"
    >
      <div
        className="Result_month_taken"
      >
        <div
          className="Result_results_description"
        >
          month
            
          <span
            className="month_taken"
          >
            06/24/2021
          </span>
        </div>
      </div>
      <div
        className="Result__meal_taken"
      >
        <div
          className="Resultresults_meal_taken"
        >
          meal
             
          <span
            className="meal_taken"
          />
        </div>
      </div>
      <div
        className="Result_result_read"
      >
        <div
          className="Result_result_read"
        >
          result
             
          <span
            className="Result_read"
          />
        </div>
      </div>
      <div
        className="Result_result_dates"
      >
        <div
          className="Result_result_date__tested"
        >
          date
             
          <span
            className="date_tested"
          >
            06/24/2021
          </span>
        </div>
      </div>
      <div
        className="Results_description"
      >
        <div
          className="Result_results_description"
        >
          Description
             
          <span
            className="Description"
          />
        </div>
      </div>
      <div
        className="Result_diabetestype"
      >
        <div
          className="Result_results_diabetestype"
        >
          diabetestype
             
          <span
            className="Result_diabetestype"
          />
        </div>
      </div>
      <button
        className="Result_delete"
        onClick={[Function]}
        type="button"
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;
